pkg/crcontrollers/drive/drivecontroller.go

```diff
@@ -455,3 +459,53 @@ func (c *Controller) removeRelatedAC(ctx context.Context, log *logrus.Entry, cur
 
 	return nil
 }
+
+// handle drive lable update
+func (c *Controller) handleDriveLableUpdate(ctx context.Context, log *logrus.Entry, drive *drivecrd.Drive) error {
+	var taintLabelExisted bool
+	labels := drive.GetLabels()
+	// if drive is taint and labeled, need to update related ac label
+	if effect, ok := labels[apiV1.DriveTaintKey]; ok && effect == apiV1.DriveTaintValue {
+		taintLabelExisted = true
+	}
+
+	// get ac location for lvg/non-lvg case
+	// lvg ac: lvg location
+	// non-lvg ac:  drive name
+	location := drive.GetName()
+	lvg, err := c.crHelper.GetLVGByDrive(ctx, drive.Spec.UUID)
+	if err != nil {
+		log.Infof("LVG for drive %s is not found: %s", drive.GetName(), err.Error())
+	} else if lvg != nil {
+		location = lvg.GetName()
+	}
+	// sync label to related ac with the drive
+	ac, err := c.crHelper.GetACByLocation(location)
+	if err != nil && err != errTypes.ErrorNotFound {
+		log.Errorf("Failed to get AC for Drive %s: %s", drive.GetName(), err.Error())
+		return err
+	}
+	if err == errTypes.ErrorNotFound {
+		log.Warnf("AC for Drive %s is not found", drive.GetName())
+		return nil
+	}
+
+	acLabels := ac.GetLabels()
+	if acLabels == nil {
+		// label is not existed on CR, just create it and attach to CR
+		acLabels = map[string]string{}
+		ac.Labels = acLabels
+	}
+	if taintLabelExisted {
+		acLabels[apiV1.DriveTaintKey] = apiV1.DriveTaintValue
+	} else {
+		delete(acLabels, apiV1.DriveTaintKey)
+	}
+
```

Comment:
Do we also need to sync this label to the corresponding lvg object for potential use?

Reply:
currently, no sync to lvg object as there is no case to use lvg to schedule.  we may add it when it is necessary.